From 38fa41222f4c03c06cae5111c0d536d7ff09ed36 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Mon, 31 Oct 2016 16:31:45 +0200
Subject: [PATCH 0557/1345] dts: a80x0: add default ECO configuration of
 A8040-DB revision A1

commit  9be9421f64215450f63c4bf9b6e1dc85a94acdec from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Updated default ECO configuration for A8040-DB board A1 revision:

SerDes & USB2/3 map:
Signed-off-by: Meng Li <Meng.Li@windriver.com>
--------------------
CP0:
Lane# | Interface
0     | PCIe0 (x1)
1     | SATA0
2     | 10G (KR)
3     | SATA1
4     | USB3_HOST1
5     | PCIe2 (x1)
USB2_H0 (UTMI only)
USB2_H1 (UTMI & USB3 SerDes)

CP1:
Lane# | Interface
0     | PCIe0 (x1)
1     | SATA0
2     | 10G (KR)
3     | SATA1
4     | PCIe1 (x1)
5     | PCIe2  x1)
USB2_H0 (UTMI only)

MPPs:
-----
AP:
	[0-10]  SDIO
	[11,19] UART0 [11,19]
CP0:
	[32,34] GE_MDIO/MDC
	[33]    GPIO: GE_INT#/push button/Wake
	[35]    MSS_GPIO[3]: MSS_PWDN
	[36]    MSS_GPIO[5]: MSS_VTT_EN
	[37-38] I2C0
	[40-41] SATA[0/1]_PRESENT_ACTIVEn
	[42-43] XG_MDC/XG_MDIO (XSMI)
	[44-55] RGMII1
	[56-62] SD

CP1:
	[0-11]  RGMII0
	[13-16] SPI1
	[27,31] GE_MDIO/MDC
	[29-30] UART0

Change-Id: I5f4a3f4c1a5326a8f35cfa7a412b8b78a2986655
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33502
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    3 +
 .../mvebu/a7k-a8k/armada8040-db-setup.txt          |  124 +++++++++------
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts   |  165 ++++++++++++++++++++
 .../boot/dts/marvell/armada-8040-db-router.dts     |    2 +-
 5 files changed, 243 insertions(+), 52 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index a00d42c..60a15aa 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -32,6 +32,9 @@ Boards:
    - "marvell,armada7040-pd"
       when the board being used is the Armada 7040 emulation platform
 
+   - "marvell,armada8040-db-A", "marvell,armada8040-db"
+      when the board being used is the Armada 8040 Development board
+
    - "marvell,armada8040-db-router", "marvell,armada8040-db"
       when the board being used is the Armada 8040 Development board
 
diff --git a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
index 275350aa..50ad4eb 100644
--- a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
@@ -19,74 +19,96 @@ To switch board configuration:
 Board Setup
 ------------
 
-## setup 1: Default ###
-
-   - Serdes CP0 site module (CON42): SLM1521
-   - Serdes CP1 site module (CON41): SLM1521
-   - Jumper-module (CON44): SLM1564
-   - SLM1521 Module DIP switch:
-	- DPR 2,3,4,5 = 1-2
-	- SW1 = ON
-	- SW2 = OFF
-	- SW3 = ON
-	- SW4 = OFF
-	- SW5 = OFF
-	- SW6 - ON
-	- SW7 - ON
-	- SW8 - OFF
+### setup A: armada-8040-db-A.dts  ###
+--------------------------------------
+
+	- Jumper-module (CON44): SLM1564
+	- Serdes CP0 site module (CON42): SLM1521:
+		- DPR 2,3,4,5 = 2-3
+		- DIP-SW-SW1:
+			SW1, SW2, SW3, SW4, SW5, SW6, SW7, SW8 = OFF
+
+	- Serdes CP1 site module (CON41): SLM1521
+		- DPR 2,3,4,5 = 2-3
+		- DIP-SW-SW1:
+			SW5 = ON
+			SW1, SW2, SW3, SW4, SW6, SW7, SW8 = OFF
+
+
+### setup 2: armada-8040-db-router.dts ###
+------------------------------------------
+
+	- Jumper-module (CON44): SLM1564
+	- Serdes CP0 site module (CON42): SLM1521
+	- Serdes CP1 site module (CON41): SLM1521
+	  SLM1521 Module DIP switch:
+		-DPR 2,3,4,5 = 2-3
+		-DIP-SW-SW1:
+			SW1, SW3, SW6, SW7 = ON
+			SW2, SW4, SW5, SW8 = OFF
 
 
 The tables below summarizes the interface configuration of each setup
 
 SERDES PHY configuration
 ------------------------
-Setup  #	|	1	|	TBD	| 	TBD	|
-Name   #	|     Default 	|		|		|
+Setup  #	|	A	|	B	| 	TBD	|
+Name   #	|     db-A	|    db-router	|		|
 -----------------------------------------------------------------
 CP0:								|
 -----------------------------------------------------------------
-	lane0	| sgmii2	|		|		|
-	lane1	| usb3h0	|		|		|
-	lane2	| KR (10G)	|		|		|
-	lane3	| sata1		|		|		|
-	lane4	| usb3h1	|		|		|
-	lane5	| pex_x1_2	|		|		|
------------------------------------------------------------------
-CP1:								|
------------------------------------------------------------------
-	lane0	| sgmii2	|		|		|
-	lane1	| usb3h0	|		|		|
-	lane2	| KR (10G)	|		|		|
-	lane3	| sata1		|		|		|
-	lane4	| N/C		|		|		|
-	lane5	| pex_x1_2	|		|		|
+	lane0	|   PCIe0 (x1)	|   N/C		|		|
+	lane1	|   SATA0	|   USB3_HOST0	|     		|
+	lane2	|   KR (10G) 	|   KR (10G)  	|		|
+	lane3	|   SATA1	|   SATA1	|		|
+	lane4	|   USB3_HOST1 	|   USB3_HOST1	|		|
+	lane5	|   PCIe2 (x1) 	|   PCIe2 (x1)	|		|
+--------------------------------|--------------------------------
+USB2_H0 (UTMI only)						|
+USB2_H1 (UTMI & USB3 SerDes)					|
+								|
+CP1:		|		| 		|		|
+--------------------------------|--------------------------------
+	lane0	|  PCIe0 (x1)	|    N/C	|		|
+	lane1	|  SATA0	|    USB3_HOST0	|		|
+	lane2	|  KR (10G)	|    KR (10G)   |		|
+	lane3	|  SATA1	|    SATA1	|		|
+	lane4	|  PCIe1 (x1)	|    USB3_HOST1	|		|
+	lane5	|  PCIe2 (x1)	|    PCIe2 (x1) |		|
 -----------------------------------------------------------------
-
+USB2_H0 (UTMI only)
 
 
 Multi-purpose pin configurations
 --------------------------------
-Setup  #	|	1	|	TBD	|	TBD	|
-Name   #	|     Default 	|		|		|
+Setup  #	|	1	|	2	|	TBD	|
+Name   #	|     Default 	|    Router	|		|
 -----------------------------------------------------------------
 AP806:								|
 -----------------------------------------------------------------
-    AP-SDIO	| N/C		|		|		|
-    AP-SPI0	| [4-5]		|		|		|
-    AP-UART0	| [11,19]	|		|		|
------------------------------------------------------------------
+    AP-SDIO	|   [0-10]	|    N/C	|		|
+    AP-SPI0	|   N/C		|    [4-5]	|		|
+    AP-UART0	|   [11,19]	|    [11,19]	|		|
+--------------------------------|--------------------------------
 CP0:								|
------------------------------------------------------------------
-    CP0-SMI	| [32,34]	|		|		|
-    CP0-XSMI	| [35-36]	|		|		|
-    CP0-TWSI	| [37-38]	|		|		|
-    CP0-SATA1	| [40]		|		|		|
-    CP0-SATA0	| [41]		|		|		|
-    CP0-RGMII1	| [44-55]	|		|		|
-    CP0-SD	| [56-62]	|		|		|
------------------------------------------------------------------
+--------------------------------|--------------------------------
+    CP0-SMI	|   [32,34]	|    [32,34]	|		|
+    CP0-XSMI	|   [42-43]	|    [35-36]	|		|
+    CP0-TWSI	|   [37-38]	|    [37-38]	|		|
+    CP0-SATA1	|   [40]	|    [40]	|		|
+    CP0-SATA0	|   [41]	|    [41]	|		|
+    CP0-RGMII1	|   [44-55]	|    [44-55]	|		|
+    CP0-SD	|   [56-62]	|    [56-62]	|		|
+--------------------------------|--------------------------------
+GPIOS:								|
+[33] - GPIO: GE_INT#/push button/Wake				|
+[35] - MSS_GPIO[3]: MSS_PWDN					|
+[36] - MSS_GPIO[5]: MSS_VTT_EN					|
+								|
 CP1:								|
------------------------------------------------------------------
-    CP1-RGMII1	| [0-11]	|		|		|
-    CP1-SMI	| [27,31]	|		|		|
+--------------------------------|--------------------------------
+    CP1-RGMII1	|    [0-11]	|    [0-11]	|		|
+    CP1-SMI	|    [27-28]	|    [27,31]	|		|
+    CP1-SPI1	|    [13-16]	|    N/C	|		|
+    CP1-UART0	|    [29-30]	|    N/C	|		|
 -----------------------------------------------------------------
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 504338f..ece76b8 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -15,6 +15,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-nand.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-all-pcie.dtb
 dtb-$(CONFIG_ARCH_MVEBU_PD) += armada-7040-pd.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-router.dtb
 
 always		:= $(dtb-y)
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
new file mode 100644
index 0000000..665286b
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 8040 development board
+ * This file supports the default configuration of the board
+ */
+
+#include "armada-8040-db.dtsi"
+
+/ {
+	model = "Marvell Armada-8040 development board router setup";
+	compatible = "marvell,armada8040-db-A", "marvell,armada8040-db", "marvell,armada8040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	chosen { };
+
+	cpn-110-master {
+		config-space {
+			sata@540000 {
+				status = "okay";
+			};
+			/* usb3h0 is USB2 only (UTMI) */
+			usb3h0: usb3@500000 {
+				status = "okay";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					phy-mode = "kr"; /* lane-2 */
+				};
+				emac2: mac2 {
+					phy-mode = "rgmii"; /* lane-0 */
+					phy = <&phy2>;
+				};
+				emac3: mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+				};
+			};
+			ppv22@000000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
+
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2: eth2@030000 {
+					status = "okay";
+				};
+			};
+			eip197@800000 {
+				status = "okay";
+			};
+		};
+		pcie@0x600000 {
+			status = "okay";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+	cpn-110-slave {
+		config-space {
+			/* usb3h0_1 is USB2 only (UTMI) */
+			usb3h0_1: usb3@500000 {
+				status = "okay";
+			};
+			sata@540000 {
+				status = "okay";
+			};
+			gop {
+				emac0_1: mac0 {
+					phy-mode = "kr"; /* lane-2 */
+				};
+				emac2_1: mac2 {
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&phy2>;
+				};
+				emac3_1: mac3 {
+					phy-mode = "sgmii"; /* lane-4 */
+				};
+			};
+			ppv22@000000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&cps_ge1_rgmii_pins>;
+
+				eth0_1: eth0@010000 {
+					status = "okay";
+				};
+				eth1_1: eth1@020000 {
+					status = "okay";
+				};
+				eth2_1: eth2@030000 {
+					status = "disabled";
+				};
+			};
+		};
+		pcie@0x600000 {
+			status = "okay";
+		};
+		pcie@0x620000 {
+			status = "okay";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
index 1df4cee..bbcf7f4 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
@@ -42,7 +42,7 @@
 
 /*
  * Device Tree file for Marvell Armada 8040 development board
- * This board file supports the router configuration of the board
+ * This board file supports the A0 revision of the board
  */
 
 #include "armada-8040-db.dtsi"
-- 
1.7.9.5

